nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -167,10 +255,32 @@ public void deleteEntry(DocumentModel docModel) throws DirectoryException {
 
     @Override
     public void deleteEntry(String id) throws DirectoryException {
+        deleteEntry(directoryName, id);
+    }
+
+    @Override
+    public void deleteEntry(String id, Map<String, String> map) throws DirectoryException {
+        deleteEntry(id);
+    }
+
+    public void deleteEntry(String collection, String id) throws DirectoryException {
         checkPermission(SecurityConstants.WRITE);
         checkDeleteConstraints(id);
+
+        for (Reference reference : getDirectory().getReferences()) {
+            if (reference instanceof MongoDBReference) {
+                MongoDBReference mongoDBReference = (MongoDBReference) reference;
+                mongoDBReference.removeLinksForSource(id, this);
+            } else {
+                reference.removeLinksForSource(id);
+            }
+        }
+
         try {
-            DeleteResult result = getCollection().deleteOne(
+            if (StringUtils.isBlank(collection)) {
```

Comment:
I'd prefer `isEmpty` instead of `isBlank` for a parameter that comes from other Java code (and not user-written config files). It's not our job here to trim spaces. Actually `collection == null` would even be better no? There's no use case for passing an empty value?